packages/select/src/components/select/select.tsx

```diff
@@ -155,7 +186,11 @@ export class Select<T> extends AbstractPureComponent2<SelectProps<T>, ISelectSta
                 {...popoverProps}
                 className={classNames(listProps.className, popoverProps.className)}
                 onInteraction={this.handlePopoverInteraction}
-                popoverClassName={classNames(Classes.SELECT_POPOVER, popoverProps.popoverClassName)}
+                popoverClassName={classNames(
+                    Classes.SELECT_POPOVER,
+                    popoverProps.popoverClassName,
+                    matchTargetWidth ? Classes.MATCH_TARGET_WIDTH : undefined,
```

Comment:
use `classNames` object API like this:

```suggestion
                    { [Classes.MATCH_TARGET_WIDTH]: matchTargetWidth },
```